runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,391 @@
+                            log.debug("[{}] {} {}", clientName, "No matching destinations found for:", removeRemoteDestinationsSet);
+                            return null;
+                        } else if (clientDestinationIdsList.size() + removeRemoteDestinationsSet.size() > clientDestinationIdsSet.size()) {
+                            removeRemoteDestinationsSet.removeAll(clientDestinationIdsSet);
+                            log.debug("[{}] {} {}", clientName, "Following destinations to remove are not present:", removeRemoteDestinationsSet);
+                        }
+
+                        if (clientDestinationIdsList.size() == 0) {
+                            txn.delete(sourceMetadataTable, clientInfoKey);
+                        } else {
+                            clientDestinations = clientDestinations.toBuilder()
+                                    .clearDestinationIds()
+                                    .addAllDestinationIds(clientDestinationIdsList)
+                                    .build();
+                            txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                        }
+
+                        txn.commit();
+                        return null;
+                    }
+                    throw new NoSuchElementException(String.format("Record not found for group: %s", logicalGroup));
+                } catch (TransactionAbortedException tae) {
+                    log.error("[{}] {}: {}", clientName, "Unable to remove destinations, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("[{}] {}: {}", clientName, "Unable to remove destinations", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Displays the destinations associated with a specific logical group.
+     *
+     * @param logicalGroup The group whose destinations will be displayed.
+     */
+    public void showDestinations(String logicalGroup) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    DestinationInfoVal clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                    if (clientDestinations != null) {
+                        List<String> clientDestinationIdsList = new ArrayList<>(clientDestinations.getDestinationIdsList());
+
+                        log.info("\n========== Destinations for {} ==========\n", logicalGroup);
```

Comment:
didnt we discuss that you are now returning a list of the destinations?  Is this an older version of the change?

Reply:
yes, I think I pushed new changes the exact moment you added comments by the looks of it